### PR TITLE
Adds update_package_properties and build_packages in pg-auto-failover-enterprise

### PR DESCRIPTION
**Pull request by gurkanindibay**
pipelines

### CODE REVIEW

**Comment by hanefi** [comment]
Should be blocked until #709 gots merged. After that one is complete, you can update this PR.

File: .github/workflows/build-package.yml
Comment (hanefi) [inline .github/workflows/build-package.yml]: needs a tag

File: .github/workflows/build-package.yml
Comment (hanefi) [inline .github/workflows/build-package.yml]: Does Xenial build properly?

File: .github/workflows/build-package.yml
Comment (hanefi) [inline .github/workflows/build-package.yml]: inconsistent use of quotations

File: .github/workflows/update_package_properties.yml
Comment (hanefi) [inline .github/workflows/update_package_properties.yml]: inconsistent use of quotations

File: .github/workflows/update_package_properties.yml
Comment (hanefi) [inline .github/workflows/update_package_properties.yml]: needs a tag

**Review by hanefi — CHANGES_REQUESTED** [review]
After addressing my comments, please compare Github Workflow definitions in `all-pgautofailover` branch with the ones in here.

File: .github/workflows/build-package.yml
Comment (gurkanindibay) [reply .github/workflows/build-package.yml]: yes

File: .github/workflows/build-package.yml
Comment (gurkanindibay) [reply .github/workflows/build-package.yml]: Fixed

File: .github/workflows/build-package.yml
Comment (gurkanindibay) [reply .github/workflows/build-package.yml]: Fixed

File: .github/workflows/update_package_properties.yml
Comment (gurkanindibay) [reply .github/workflows/update_package_properties.yml]: Fixed

File: .github/workflows/update_package_properties.yml
Comment (gurkanindibay) [reply .github/workflows/update_package_properties.yml]: Fixed

File: .github/workflows/build-package.yml
Comment (hanefi) [inline .github/workflows/build-package.yml]: inconsistent use of quotation marks.

`--platform "${{ matrix.platform }}"`

or 

`--platform ${{ matrix.platform }}`

make up your mind, and stick to it.

File: .github/workflows/build-package.yml
Comment (hanefi) [inline .github/workflows/build-package.yml]:
```suggestion
        run: git clone -b v0.8.1 --depth=1 https://github.com/citusdata/tools.git tools
```

File: .github/workflows/build-package.yml
Comment (hanefi) [inline .github/workflows/build-package.yml]:
```suggestion
          python -m tools.packaging_automation.citus_package \
```

File: .github/workflows/build-package.yml
Comment (hanefi) [inline .github/workflows/build-package.yml]:
```suggestion
          python -m tools.packaging_automation.upload_to_package_cloud \
```

File: .github/workflows/update_package_properties.yml
Comment (hanefi) [inline .github/workflows/update_package_properties.yml]:
```suggestion
        run: git clone -b v0.8.1 --depth=1 https://github.com/citusdata/tools.git tools
```

File: .github/workflows/build-package.yml
Comment (hanefi) [inline .github/workflows/build-package.yml]:
```suggestion
    branches:
    - "**"
```

File: .github/workflows/build-package.yml
Comment (hanefi) [inline .github/workflows/build-package.yml:6]: IIRC, Github recommends that all secrets should be stored in env variables, just like this one here.

Does that make sense to do? What do you think?

File: .github/workflows/build-package.yml
Comment (hanefi) [reply .github/workflows/build-package.yml]: this is a surprise to me as we were not able to build packages on PGAF community

**Review by hanefi — CHANGES_REQUESTED** [review]

File: .github/workflows/build-package.yml
Comment (hanefi) [reply .github/workflows/build-package.yml]: I do not think so. Please take a look at my other comments.

File: .github/workflows/update_package_properties.yml
Comment (hanefi) [reply .github/workflows/update_package_properties.yml]: I do not think so. Please take a look at my other comments.

File: .github/workflows/build-package.yml
Comment (gurkanindibay) [reply .github/workflows/build-package.yml]: it also failed this time I needed to remove it

File: .github/workflows/build-package.yml
Comment (hanefi) [reply .github/workflows/build-package.yml:6]: This comment affect all branches where we started to use GH actions. I opened an issue for this task: https://github.com/citusdata/packaging/issues/740

File: .github/workflows/build-package.yml
Comment (gurkanindibay) [reply .github/workflows/build-package.yml]: github variables don't need to be in quotation marks. So I don't put quotation marks.For environment variables quotation marks are needed

File: .github/workflows/build-package.yml
Comment (gurkanindibay) [reply .github/workflows/build-package.yml:6]: I will check all of them I missed this oneç Normally I'm trying to be careful
Fixed this one

File: .github/workflows/build-package.yml
Comment (hanefi) [inline .github/workflows/build-package.yml]:
```suggestion
          --build_type "release" \
```

File: .github/workflows/update_package_properties.yml
Comment (hanefi) [inline .github/workflows/update_package_properties.yml:38]: https://github.com/citusdata/packaging/pull/713#discussion_r696468145 :

> Github variables don't need to be in quotation marks. So I don't put quotation marks. For environment variables quotation marks are needed

Does that mean, that you should have the following here:
```suggestion
          git config --global user.email ${{ github.event.inputs.microsoft_email }} && \
          git config --global user.name ${{ github.event.inputs.name }}
```

I do not like or promote your idea on when to use double quotes. I think that we should always use double quote to prevent globbing and word splitting. See: https://github.com/koalaman/shellcheck/wiki/SC2086

For the time being, I am ok with anything as long as you are consistent and have a policy about this.

File: .github/workflows/build-package.yml
Comment (gurkanindibay) [reply .github/workflows/build-package.yml]: I looked into a little bit and using double quotes for Github Actions parameters as a standard is better way in bash scripts. Changing like this

File: .github/workflows/build-package.yml
Comment (gurkanindibay) [reply .github/workflows/build-package.yml]: Fixed as you suggested

File: .github/workflows/update_package_properties.yml
Comment (gurkanindibay) [reply .github/workflows/update_package_properties.yml:38]: Used double quotes as a standard

File: .github/workflows/update_package_properties.yml
Comment (gurkanindibay) [reply .github/workflows/update_package_properties.yml]: Used double quotes as standard

File: .github/workflows/build-package.yml
Comment (gurkanindibay) [reply .github/workflows/build-package.yml]: Used double quotes as standard

File: .github/workflows/build-package.yml
Comment (hanefi) [inline .github/workflows/build-package.yml]:
```suggestion
        run: python -m pip install -r tools/packaging_automation/requirements.txt
```

no need to do this again.

**Review by hanefi — COMMENTED** [review]
Let's remove ubuntu/xenial from travis tests as well

**Review by hanefi — APPROVED** [review]

File: .github/workflows/build-package.yml
Comment (gurkanindibay) [reply .github/workflows/build-package.yml]: Fixed

**Pull request merged**
